packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -1709,49 +1755,73 @@ class _NavigationBarComponentsTransition {
     );
   }
 
-  // Create a Tween that moves a widget between its original position in its
-  // ancestor navigation bar to another widget's position in that widget's
-  // navigation bar.
+  // Create an animated widget that moves the given child widget between its
+  // original position in its ancestor navigation bar to another widget's
+  // position in that widget's navigation bar.
   //
   // Anchor their positions based on the vertical middle of their respective
   // render boxes' leading edge.
   //
-  // Also produce RelativeRects with sizes that would preserve the constant
-  // BoxConstraints of the 'from' widget so that animating font sizes etc don't
-  // produce rounding error artifacts with a linearly resizing rect.
-  RelativeRectTween slideFromLeadingEdge({
+  // This method assumes there's no other transforms other than translations
```

Comment:
Going through the bug, it seems like this wasn't the issue. Would still be nice to address separately (since the beginning state isn't wrapped, the end state isn't wrapped either, but the transition might be wrapped). But otherwise the PR LGTM to fix the rounding issue

Reply:
During the animation the font size can change, and in the case where the font size gets larger but the container is still able to provide enough horizontal space the text shouldn't wrap?

As I understand it during the animation the middle widget has a start size but does not typically have an end size:
```
      return slideFromLeadingEdge(
        fromKey: bottomComponents.middleKey,
        fromNavBarBox: bottomNavBarBox,
        toKey: topComponents.backLabelKey,
        toNavBarBox: topNavBarBox,
``` 
So the middle widget moves to align with the backLabelKey when the animation completes and it's not given a size.